src/main/java/org/datavyu/plugins/nativeosx/AVFoundationMediaPlayer.java

```diff
@@ -12,51 +14,76 @@
 
 public class AVFoundationMediaPlayer extends NativeOSXMediaPlayer {
 
+  private static Logger logger = LogManager.getLogger(AVFoundationMediaPlayer.class);
+
   private Container container;
   private final Object readyLock = new Object();
 
+  private long startInitTime;
+  private long endInitTime;
+
   public AVFoundationMediaPlayer(URI mediaPath, Container container) {
     super(mediaPath);
     this.addMediaPlayerStateListener(new AVFoundationMediaPlayer.PlayerStateListenerImpl());
     this.container = container;
+    sendPlayerStateEvent(eventPlayerUnknown, 0);
   }
 
   @Override
   public void init() {
+    logger.info("Initializing " + this.getClass().getName());
+
+    startInitTime = System.currentTimeMillis();
+
     initNative();
 
-    // Test if he file exists
+    // Test if the file exists
     File mediaFile = new File(mediaPath);
     if (!mediaFile.exists()){
       throwMediaErrorException(MediaError.ERROR_MEDIA_INVALID.code(), null);
     }
 
     container.addNotify();
-    container.add(nativePlayerCanvas, BorderLayout.CENTER);
+    container.add(mediaPlayer, BorderLayout.CENTER);
 
     incPlayerCount();
-    sendPlayerStateEvent(eventPlayerReady, 0);
+    Runnable waitForReady = () -> {
```

Comment:
Can we design this different by listening to some events?

Reply:
Unfortunately, there is no other way to solve the fps problem from the java side. The only way to solve it is to block in the native side during the initialization. The current implementation of the AVFoundation player doesn't propagate events or errors to the java side

Reply:
OK